Source/ProgressionSystemRuntime/Public/Widgets/PSMenuWidget.h

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Valerii Rotermel & Yevhenii Selivanov
+
+#pragma once
+
+#include "Blueprint/UserWidget.h"
+#include "PoolManagerTypes.h"
+#include "PSMenuWidget.generated.h"
+
+/**
+ * Widget to display the progression as stars in the main menu.
+ */
+UCLASS()
+class PROGRESSIONSYSTEMRUNTIME_API UPSMenuWidget : public UUserWidget
+{
+	GENERATED_BODY()
+
+public:
+	/**
+	 * Dynamically populates a Horizontal Box with images representing unlocked and locked progression icons.
+	 * @param AmountOfUnlockedPoints The number of images (unlocked-icon as images) to be displayed 
+	 * @param AmountOfLockedPoints The number of images (locked-icon as images) to be displayed
+	 */
+	UFUNCTION(BlueprintCallable, Category= "C++")
+	void AddImagesToHorizontalBox(float AmountOfUnlockedPoints, float AmountOfLockedPoints);
+
+	/**
+	* Sets the visibility of the background overlay and lock icon.
+	* @param VisibilitySlate The visibility state (e.g., Visible, Collapsed) to apply to the overlay and icon.
+	*/
+	UFUNCTION(BlueprintCallable, Category= "C++", meta = (BlueprintProtected))
+	void SetOverlayVisibility(ESlateVisibility VisibilitySlate);
+
+	/*********************************************************************************************
+	 * Protected functions
+	 ********************************************************************************************* */
+protected:
+	// Horizontal Box widget for storing stars
+	UPROPERTY(EditDefaultsOnly, Category = "C++", meta = (BlueprintProtected, BindWidget))
+	TObjectPtr<class UHorizontalBox> HorizontalBox = nullptr;
+
+	/** Background overlay for tint effect */
+	UPROPERTY(EditDefaultsOnly, Category = "C++", meta = (BlueprintProtected, BindWidget))
+	TObjectPtr<class UImage> PSCBackgroundOverlay = nullptr;
+
+	/** Background overlay lock icon */
+	UPROPERTY(EditDefaultsOnly, Category = "C++", meta = (BlueprintProtected, BindWidget))
```

Comment:
Every `BindWidget` property has to have `Transient` specified, see `As of Unreal Engine 5.0 you must mark your properties as Transient` from [link](https://benui.ca/unreal/ui-animations-from-cpp/)
Also, add BlueprintReadWrite to be able to access it in blueprints.
Do it for all your `BindWidget` properties in both PSMenu and PSStart widgets

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/1/commits/cbf5176e1009f23a1bdea1f9d7b7dad1294d391a

Reply:
@h4rdmol Is not finished. You completely ignored next from above:
| Also, add BlueprintReadWrite to be able to access it in blueprints.

Reply:
added https://github.com/h4rdmol/ProgressionSystem/pull/1/commits/3ba812e2524bd739f0dde7c5c8ba7c6341a0a62f

Reply:
@h4rdmol, is not finished yet. You added BlueprintReadWrite as 'meta' inside `meta = ()`, removing `BlueprintProtected`. It's wrong and will not work. It should look like next:
```
UPROPERTY(EditDefaultsOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, BindWidget))
```
But now you have:
```
UPROPERTY(EditDefaultsOnly, Transient, Category = "C++", meta = (BlueprintReadWrite, BindWidget))
```
Just find any example of my code in the project

Reply:
got it, thank you https://github.com/h4rdmol/ProgressionSystem/pull/1/commits/83a38896b1f64d6c753723373949a735bc0ebf3e